packages/engine-server/src/drivers/file/storev2.ts

```diff
@@ -146,9 +147,27 @@ export class FileStorage implements DStore {
     }
   }
 
-  private static createMalformedSchemaError(resp: DEngineInitSchemaResp) {
+  static createMalformedSchemaError(resp: DEngineInitSchemaResp) {
+    let filePath = USER_MESSAGES.UNKNOWN;
+    try {
+      if (resp.error && resp.error.payload) {
+        filePath = JSON.parse(JSON.parse(resp.error.payload)[0].payload).fpath;
+      }
+    } catch (parseErr) {
+      filePath = USER_MESSAGES.UNKNOWN;
+    }
+
+    let reason = USER_MESSAGES.UNKNOWN;
+    try {
+      if (resp.error && resp.error.payload) {
+        reason = JSON.parse(JSON.parse(resp.error.payload)[0].payload).message;
+      }
+    } catch (parseErr) {
+      reason = USER_MESSAGES.UNKNOWN;
+    }
+
     return new DendronError({
-      message: "schema malformed",
+      message: `Schema '${filePath}' is malformed. Reason: ${reason}. Use 'Go to File...' command to resolve malformed schema.`,
```

Comment:
Speaking out loud - it'd be nice for us to include a button in the error notification such that they can click on it and be taken directly to the schema file for editing.  It'd be hard (and perhaps undesirable) to pack that info into a DendronError though. Something maybe for V2/V3

Reply:
Something like optional `errorData.fileAtFault` seems reasonable to have within DendronError. Dendron deals with files so fileAtFault in DendronError does not appear to break placement. Not sure if we can add buttons with `window.showErrorMessage` though.

Reply:
It should be possible with the `items[]` param: https://code.visualstudio.com/api/references/vscode-api#window.showErrorMessage

Reply:
Added a follow up ![[scratch.2021.11.04.134933.add-file-navigation-in-schema-malformed-message]]